pkg/build/builder/docker.go

```diff
@@ -66,7 +68,14 @@ func (d *DockerBuilder) Build() error {
 	if err != nil {
 		return err
 	}
-	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, initialURLCheckTimeout, os.Stdin, d.gitClient)
+	startTime := unversioned.Now()
+	sourceInfo, stages, err := fetchSource(d.dockerClient, buildDir, d.build, initialURLCheckTimeout, os.Stdin, d.gitClient)
+	stopTime := unversioned.Now()
+	var subStepDuration int64
+	d.build.Status.Stages, subStepDuration = api.MergeStageAndStepInfo(d.build.Status.Stages, stages)
+	stepDuration := int64(stopTime.Time.Sub(startTime.Time)) - subStepDuration
```

Comment:
at a minimum this needs some comments but fundamentally it seems like we're doing more manipulation than i'd expect...what's going on here?